core/federated/network/socket_common.c

```diff
@@ -235,7 +235,7 @@ int connect_to_socket(int sock, const char* hostname, int port) {
     }
     // Convert port number to string.
     char str[6];
-    sprintf(str, "%u", used_port);
+    snprintf(str, sizeof(str), "%u", used_port);
```

Comment:
This doesn't look right to me.  `str` is a pointer, so `sizeof(str)` is the size of a pointer, which is 8 bytes on a 64-bit machine. This will cause a buffer overrun.

Reply:
Hmm.  I couldn't find anything more authoritative about this. Seems brittle.  Would it be better to use a `#define` instead?

Reply:
The behavior `sizeof` operator in the C language is well defined. It universally works as it's defined in the C language standard - thus, it is nothing hacky, brittle, or platform-dependent. This may be a more authoritative document: https://learn.microsoft.com/en-us/cpp/c-language/sizeof-operator-c?view=msvc-170 

Using #define constant like #define PORT_STR_SIZE 6 also works. Still, I believe it's safer to do:
```
#define PORT_STR_SIZE 6
char str[PORT_STR_SIZE];
snprintf(str, sizeof(str), "%u", used_port);
```
than doing:
```
#define PORT_STR_SIZE 6
char str[PORT_STR_SIZE];
snprintf(str, PORT_STR_SIZE, "%u", used_port);
```
I'm fine with either option above, with a preference to the former.